pyrobuf/parse_proto.py

```diff
@@ -231,15 +232,29 @@ def parse(
                 )
 
             elif token.token_type == 'MESSAGE':
-                rep['messages'].append(
-                    self._parse_message(
-                        token, tokens, messages, enums.copy(), imported['enums']
-                    )
+                ret = self._parse_message(
+                        token, tokens, messages, enums.copy(),
+                        imported['enums']
                 )
 
+                assert token.name not in scope, (
+                    "'{}' is already defined in global scope".format(
+                        token.name))
+
+                scope[token.name] = ret
+                rep['messages'].append(ret)
+
             elif token.token_type == 'ENUM':
-                ret = self._parse_enum(token, tokens)
-                rep['enums'].append(ret)
+                try:
+                    rep['enums'].append(
+                        self._parse_enum(token, tokens, scope)
+                    )
+                except AssertionError as e:
+                    message, *rest = e.args
+                    message += " in global scope"
```

Comment:
This feels a little sketchy - what if we add other assertions in `_parse_enum` later? I'd prefer to just remove the `try/except` block and lose the global scope info. Maybe we could add some sort of scope tag to the scope dict so that we can add appropriate scope info on the initial assertion?